schema-layer/data-structures/data.md

```diff
@@ -0,0 +1,43 @@
+`Data` is an advanced layout for representing binary data.
+
+It is flexible enough to support very small and very large (multi-block) binary data.
+
+```sh
+type Lengths [Int]
+type ByteUnionList [&BytesUnion]
+
+type NestedByteList struct {
+  lengths Lengths
+  parts ByteUnionList
+  algo optional String
+}
+
+type BytesUnion union {
+  | Bytes bytes
+  | &Bytes link
```

Comment:
what purpose does the link serve now? Since `ByteUnionList` is a list of links, so it could be made up of links to links to bytes? I think the only thing I can see it being useful for here is for the case of being able to use `raw` blocks--so a parent block just has `size:Int` and `bytes:CID` and following the CID gives you a raw block of bytes. Is that what this is for? How much do we need to care about that case?

Reply:
if you look closely at the new spec, this is the *only* way to link to raw blocks and not embed bytes in the CBOR block. we definitely need this ;)

Reply:
actually... maybe i’m wrong. since you can link to the kinded union you could just link to blocks.

i guess i hadn’t updated my mental model of this to the kinded union and was still thinking in terms of the keyed union. i’ll update.